Support `MessageSpec.extraConfirmText` for high-impact confirm prompts

jskupsik opened the pull request:
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

jskupsik commented:
Toolbox PR: https://github.com/xh/toolbox/pull/776

lbwexler commented:
@jskupsik -- I love that you are doing this and had been thinking this recently -- in fact I have a PR for another change to our pop-up infrastructure:

https://github.com/xh/hoist-react/pull/4054


This seems to me an elegant solution with the current API, but I am wondering if it would make sense to take it one step farther, to have MessageSpec include a 'extraConfirmText' and an `extraConfirmLabel`.  Advantages would be:

- Would be easy to document on MessageSpec, (instead of a hidden utility function)
- Could be specified in addition to an input. (not sure how common this would be).
- Could appear in a different place/color/size than the input, if our base app container wanted to make it look different than a standard prompt input.

jskupsik commented:
- Enhanced the CHANGELOG message to better fit our style.
- Added `extraConfirmText` and `extraConfirmLabel` properties to MessageSpec.
- Implemented as additional extraConfirm field on the message's formModel, so it can be combined with input if so desired.
- Changed all new usages of it to follow the new format.

lbwexler commented on desktop/appcontainer/Message.ts:
```diff
@@ -35,32 +35,50 @@ export const message = hoistCmp.factory({
 
-const inputCmp = hoistCmp.factory<MessageModel>(({model}) => {
-    const {formModel, input} = model;
+const inputsCmp = hoistCmp.factory<MessageModel>(({model}) => {
```
this is the desktop file -- what about the mobile app container?

amcclain reviewed (approved):
👍👍
